### PR TITLE
Suse Manager Settings - Frontend refactoring

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

This PR removes from redux state/sagas the suse manager settings.
This pr creates a custom hook to handle the lifecycle of suse manager settings.

UI/UX Interactions not changed.

## How was this tested?

Automated tests

### CODE REVIEW

**Review by dottorblaster — APPROVED** [review]
I can't find anything wrong with this PR :smile: 

Just for future viewers, we decided to have custom hooks in this page but it should really be taken with a grain of salt and we still don't want to adopt them in the rest of the app for now

File: assets/js/pages/SettingsPage/hooks.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+export const useSuseManagerSettings = () => {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(false);
+  const [settings, setSettings] = useState({});
+  const [entityErrors, setEntityErrors] = useState([]);
+  const [fetchError, setFetchError] = useState(false);
+  const [testingSettings, setTestingSettings] = useState(false);
+
+  const fetchSuseManagerSettings = async () => {
```
Comment (EMaksy) [inline assets/js/pages/SettingsPage/hooks.js:23]: This is very clean, i like it :D

File: assets/js/pages/SettingsPage/hooks.test.jsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+    expect(store.getActions()).toEqual([
+      {
+        type: 'NOTIFICATION',
+        payload: { text: `Unable to clear settings`, icon: '❌' },
+      },
+    ]);
+  });
+
+  it('should test the suse manager settings and spawn a toast when the connection succeded', async () => {
```
Comment (EMaksy) [inline assets/js/pages/SettingsPage/hooks.test.jsx:183]: typo in succeded -->  succeeded :+1:

**Review by EMaksy — APPROVED** [review]
Hey nice great work man, just one typo :+1:

**Pull request merged**